packages/common/context/src/context.ts

```diff
@@ -111,16 +175,26 @@ export class Context {
    */
   onDispose(callback: DisposeCallback): () => void {
     if (this.#isDisposed) {
+      let reason = ContextDisposeReason.DISPOSED;
```

Comment:
should it we maybe instead save the actual reason when a context is disposed? I mean if a context gets disposed 1ms before deadline and then onDispose gets called 2 ms after, t will be invoked with DEADLINE_EXCEEDED, even though it was DISPOSED.